runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -621,13 +651,59 @@ InterpreterEmulator::visitInvokedynamic()
 
       TR_CallSite *callsite = new (comp()->trHeapMemory()) TR_J9MethodHandleCallSite(_calltarget->_calleeMethod, callNodeTreeTop,   parent,
                                                                         callNode, interfaceMethod, resolvedMethod->classOfMethod(),
-                                                                        -1, cpIndex, resolvedMethod,
+                                                                        -1, callSiteIndex, resolvedMethod,
                                                                         resolvedSymbol, isIndirectCall, isInterface, *_newBCInfo, comp(),
                                                                         _recursionDepth, allconsts);
 
       findTargetAndUpdateInfoForCallsite(callsite);
       }
+#endif //J9VM_OPT_OPENJDK_METHODHANDLE
+   }
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+void
+InterpreterEmulator::visitInvokehandle()
+   {
+
+   bool isInterface = false;
+   bool isIndirectCall = false;
+   TR::Method *interfaceMethod = 0;
+   TR::TreeTop *callNodeTreeTop = 0;
+   TR::Node *parent = 0;
+   TR::Node *callNode = 0;
+   TR::ResolvedMethodSymbol *resolvedSymbol = 0;
+   Operand *result = NULL;
+   int32_t cpIndex = next2Bytes();
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+   if (owningMethod->isUnresolvedMethodTypeTableEntry(cpIndex)) return; // unresolved
+
+   // add appendix object to knot and push to stack
+   TR::KnownObjectTable *knot = comp()->getOrCreateKnownObjectTable();
+   uintptr_t appendixAddress = owningMethod->appendixAddressFromInvokeHandleSideTable(cpIndex);
+   if (knot) push(new (trStackMemory()) KnownObjOperand(knot->getOrCreateIndexAt(&appendixAddress)));
```

Comment:
If `knot` is null, we still need to push an operand (unknown operand) to maintain the right stack shape.